File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline  

stack = []
def push(x):
    stack.append(x)
    
def pop():
    if len(stack) > 0:
        return stack.pop()
    return -1

def size():
    return len(stack)

def empty():
    return 1 if len(stack) == 0 else 0

def top():
    if len(stack) > 0:
        return stack[-1]
    return -1

n = int(input())

for _ in range(n):
    command = input().split()
    
    if command[0] == 'push':
        push(int(command[1]))
    elif command[0] == 'pop':
        print(pop())
    elif command[0] == 'size':
        print(size())
    elif command[0] == 'empty':
        print(empty())
    elif command[0] == 'top':
        print(top())

File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const insput = fs.readFileSync(0).toString().trim().split('\n');
const [lines, ...cmds] = insput;

let newList = [];
for (let i = 0; i < cmds.length; i++) {
	let [cmd, num] = cmds[i].split(' ');
	//console.log(cmd, num);
	if (cmd == 'push') {
		newList.push(num);
	} else if (cmd == 'pop') {
		if (newList.length != 0) {
			console.log(newList[newList.length - 1]);
			newList.pop();
		} else {
			console.log(-1);
		}
	} else if (cmd == 'size') {
		console.log(newList.length);
	} else if (cmd == 'empty') {
		if (newList.length == 0) {
			console.log(1);
		} else {
			console.log(0);
		}
	} else if (cmd == 'top') {
		if (newList.length == 0) {
			console.log(-1);
		} else {
			console.log(newList[newList.length - 1]);
		}
	}
	//console.log(newList);
}
